tests/dualtor/test_link_drop.py

```diff
@@ -0,0 +1,233 @@
+import logging
+import json
+import pytest
+import tabulate
+import time
+
+from tests.common.dualtor.control_plane_utils import verify_tor_states
+from tests.common.dualtor.dual_tor_utils import upper_tor_host
+from tests.common.dualtor.dual_tor_utils import lower_tor_host
+from tests.common.dualtor.data_plane_utils import send_server_to_t1_with_action
+from tests.common.dualtor.data_plane_utils import send_t1_to_server_with_action
+from tests.common.dualtor.mux_simulator_control import set_drop
+from tests.common.dualtor.mux_simulator_control import set_output
+from tests.common.dualtor.mux_simulator_control import simulator_flap_counter
+from tests.common.dualtor.mux_simulator_control import toggle_all_simulator_ports_to_upper_tor
+from tests.common.fixtures.ptfhost_utils import run_icmp_responder
+from tests.common.fixtures.ptfhost_utils import change_mac_addresses             # lgtm[py/unused-import]
+from tests.common.fixtures.ptfhost_utils import copy_ptftests_directory
+
+
+def _set_drop_factory(set_drop_func, direction, tor_mux_intfs):
+    """Factory to get set drop function for either upper_tor or lower_tor."""
+    def _set_drop_all_interfaces():
+        logging.debug("Start set drop for %s at %s", direction, time.time())
+        for intf in tor_mux_intfs:
+            set_drop_func(intf, [direction])
+    return _set_drop_all_interfaces
+
+
+@pytest.fixture(scope="function")
+def drop_flow_upper_tor(set_drop, set_output, tor_mux_intfs):
+    """Drop the flow to the upper ToR."""
+    direction = "upper_tor"
+    yield _set_drop_factory(set_drop, direction, tor_mux_intfs)
+
+    for intf in tor_mux_intfs:
```

Comment:
`set_drop` already cleans up after itself by restoring output, could you just change the scope to function and use it directly?

Reply:
Nice catch, will update.

Reply:
`set_drop` is a module-level fixture while the cleanup needs to be done for every test function, so let's keep this function-level restoration, and let `set_drop` be our last resort to do a double cleanup.

Reply:
Sorry I meant change the scope of `set_drop` to function scope so it cleans up after every individual test. This way other tests that also use the fixture do not need to worry about function-level cleanup.